src/clj/puppetlabs/services/legacy_routes/legacy_routes_service.clj

```diff
@@ -27,7 +28,9 @@
           master-route-config (master-core/get-master-route-config
                                 master-ns
                                 config)
-          master-route-handler (-> (master-core/root-routes handle-request)
+          jruby-service (tk-services/get-service this :JRubyPuppetService)
+          master-route-handler (-> (master-core/root-routes handle-request
+                                                            jruby-service)
```

Comment:
Yeah, I'm not thrilled about this either.  I suppose we could do this differently by having a layered form of `root-routes` that the "real" master service would call with a jruby-service reference vs. one that doesn't require a jruby-service that the legacy routes service would use.  A little earlier on, we had some differences between the forms of root-routes that the master vs. legacy routes would call and that led to some problems with some aspects of the routing working differently between the two.  Can't remember the details now - maybe was tk-authz related?  In any event, I'd prefer to stick with this approach for now to avoid possibility of further divergence between the two services' uses of the master routes.  Still hoping we can kill this service soon-ish.